include/boost/geometry/algorithms/detail/buffer/turn_in_piece_visitor.hpp

```diff
@@ -127,85 +122,115 @@ inline bool in_box(Point const& previous,
     {
-        // Collinear, only on segment if it is covered by its bbox
-        if (in_box(previous, current, turn.robust_point))
+        typedef geometry::model::referring_segment<Point const> segment_type;
+        segment_type const p(turn.rob_pi, turn.rob_pj);
+        segment_type const q(turn.rob_qi, turn.rob_qj);
+        segment_type const r(previous, current);
+        int const side = strategy::side::side_of_intersection::apply(p, q, r,
+                    turn.robust_point);
+
+        if (side == 0)
         {
             return analyse_on_offsetted;
         }
+        if (side == -1 && from_monotonic)
+        {
+            return analyse_within;
+        }
+        if (side == 1 && from_monotonic)
+        {
+            return analyse_disjoint;
+        }
+        return analyse_continue;
     }
-    else if (twice_area < 0)
+};
+
+template <>
+struct check_segment<false>
+{
+    template <typename Point, typename Turn>
+    static inline analyse_result apply(Point const& previous,
+            Point const& current, Turn const& turn,
+            bool from_monotonic)
     {
-        // It is in the triangle right-of the segment where the
-        // segment is the hypothenusa. Check if it is close
-        // (within rounding-area)
-        if (twice_area * twice_area < geometry::comparable_distance(previous, current)
-            && in_box(previous, current, turn.robust_point))
+        typedef typename strategy::side::services::default_strategy
+            <
+                typename cs_tag<Point>::type
+            >::type side_strategy;
+
+        int const side = side_strategy::apply(previous, current, turn.robust_point);
+
+        if (side == 0)
         {
-            return analyse_near_offsetted;
+            // Collinear, only on segment if it is covered by its bbox
+            if (in_box(previous, current, turn.robust_point))
+            {
+                return analyse_on_offsetted;
+            }
+        }
+        else if (side == -1)
+        {
+            // It is in the triangle right-of the segment where the
+            // segment is the hypothenusa. Check if it is close
+            // (within rounding-area)
+            if (in_box(previous, current, turn.robust_point))
+            {
+                return analyse_near_offsetted;
+            }
+            else if (from_monotonic)
+            {
+                return analyse_within;
+            }
         }
         else if (from_monotonic)
         {
-            return analyse_within;
+            // Left of segment
+            return analyse_disjoint;
         }
-    }
-    else if (twice_area > 0 && from_monotonic)
-    {
-        // Left of segment
-        return analyse_disjoint;
-    }
 
-    // Not monotonic, on left or right side: continue analysing
-    return analyse_continue;
-#endif
-}
+        // Not monotonic, on left or right side: continue analysing
+        return analyse_continue;
+    }
+};
```

Comment:
Tabs. Do you have an idea why does github display the `;` differently here?

Reply:
; differently might be caused by me, splitting a free function to a member function of a class, and that might also confuse you for the 'tabs'. Because as far as I can see they are all right.